tests/app/main/test_frameworks.py

```diff
@@ -2805,14 +2810,188 @@ def test_get_with_with_previous_answers(self, data_api_client):
             doc = html.fromstring(res.get_data(as_text=True))
             assert len(doc.xpath('//input[@id="input-PR1-no"]/@checked')) == 1
 
-    def test_post_valid_data(self, data_api_client):
+    def test_get_with_with_prefilled_answers(self, data_api_client):
         with self.app.test_client():
             self.login()
+            # First call is for the current framework; second call for the framework to pre-fill from
+            data_api_client.get_framework.side_effect = [
+                self.framework(slug='g-cloud-9', name='G-Cloud 9', status='open'),
+                self.framework(slug='digital-outcomes-and-specialists-2', name='Digital Stuff 2', status='live'),
+            ]
+            # Current framework application information
+            data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(
+                framework_slug="g-cloud-9",
+                declaration={"status": "started"},
+                prefill_declaration_from_framework_slug="digital-outcomes-and-specialists-2"
+            )
+            # The previous declaration to prefill from
+            data_api_client.get_supplier_declaration.return_value = {
```

Comment:
I would say we care about *all* these conditions *all* the time. "Separate tests for separate concerns" *sounds* nice, but whichever way you look at it, doesn't end up testing as much. Only if this exact function-under-test call was made with the exact same setup prologue would it be not worth making these extra assertions here. And if that were the case, then I would advocate merging those tests to improve run time.

This comes back to the coverage argument. We're kidding ourselves with any coverage reports we're doing if we're not actually asserting all that we could be in all cases.

e.g. from what I can tell, nowhere else in the tests do we test the case where the function under test is called with the declaration being `started`, not having any pre-existing answers and having the `prefill_declaration_from_framework_slug` set. We'd be missing out on the opportunity to ensure that combination doesn't go crazy and idunno call `get_supplier_declaration` for `g-cloud-9` in this case. And we do *care* that it doesn't do that as if it doesn't call it with the right argument, there's no way it can possibly give us the right answer under real life conditions...